### PR TITLE
Senior design

### PR DESCRIPTION
**Pull request by satorinin**
This should be ready to go. Added X525 frame, planar thrust actuator, and made the simulator a bit more extendable.

### CODE REVIEW

**Comment by satorinin** [comment]
I'll resolve conflicts first. Didn't see them initially.

**Comment by satorinin** [comment]
This is ready to be reviewed

File: param/morse.yaml [hunk 1]
```diff
@@ -45,3 +45,5 @@ sim:
     # bottom_camera_resolution: [1280, 960, 60]
 
     create_teleport_actuator: false
+
```
Comment (asaba96) [inline param/morse.yaml:48]: Can we get the same comment as on line 37 in morse-prototype.yaml here?

File: param/morse-prototype.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ground_truth_camera_localization: false
+    ground_truth_roombas: true
+    ground_truth_obstacles: false
+
+    # Because MORSE spits out messages all at the same time, if you want these
+    # to come out at MORSE's update frequency (60Hz by default), you have to
+    # make this a little larger than MORSE's frequency
+    max_roomba_output_freq: 61
+    max_obstacle_output_freq: 61
+
```
Comment (asaba96) [inline param/morse-prototype.yaml:26]: I believe this is missing some params related to noisy roombas. Not sure how them missing from here will impact the simulator when the senior design drone sim is launched.

File: sim/default.py [hunk 3]
```diff
@@ -14,18 +14,21 @@
...
 create_teleport_actuator = rospy.get_param('sim/create_teleport_actuator', False)
+prototype_uav = rospy.get_param('sim/prototype_uav', True)
+num_roombas = rospy.get_param('sim/num_roombas',True)
+num_obstacle_roombas = rospy.get_param('sim/num_obstacles',True)
 
-num_targets = rospy.get_param('sim/num_roombas')
-num_obstacles = rospy.get_param('sim/num_obstacles')
 
+num_targets = num_roombas
+num_obstacles = num_obstacle_roombas
```
Comment (asaba96) [inline sim/default.py]: Was there a reason to change the way the variables `num_targets` and `num_obstacles` are initialized?

**Review by amiller27 — CHANGES_REQUESTED** [review]
Yeah this looks like it didn't merge master correctly

File: sim/default.py (hunk 3 above)
Comment (satorinin) [reply sim/default.py]: I did this so we aren't getting the sim/num_roombas parameter twice

File: param/morse.yaml (hunk 1 above)
Comment (satorinin) [reply param/morse.yaml:48]: fixed in 7af24a5

File: param/morse-prototype.yaml (hunk 2 above)
Comment (satorinin) [reply param/morse-prototype.yaml:26]: fixed in 509263d

**Comment by satorinin** [comment]
The requested changes were done, and the branch was rebased.

File: src/simulator_adapter.py [hunk 4]
```diff
@@ -316,6 +328,9 @@ def obstacle_odom_callback(msg, topic, data={}):
     bottom_camera_on = bool(rospy.get_param(
             '/sim/bottom_camera_resolution', False))
 
+    publish_prototype_uav = bool(rospy.get_param(
```
Comment (amiller27) [inline src/simulator_adapter.py]: Looks like this isn't used?

File: sim/default.py (hunk 3 above)
Comment (amiller27) [reply sim/default.py]: Why can't we just set `num_targets` and `num_obstacles` like they were before?  I think we both added the same feature and the merge messed it up

File: launch/morse.launch [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
 
-    <node name="simulator_adapter" pkg="iarc7_simulator" type="simulator_adapter.py" />
+        <node name="simulator_adapter" pkg="iarc7_simulator" type="simulator_adapter.py" />
+    </group>
+
+    <group if="$(eval str(arg('platform')) == 'sim_1.5')" >
+        <rosparam command="load" file="$(find iarc7_simulator)/param/morse-prototype.yaml" />
+        <include file="$(find iarc7_simulator)/launch/static_transforms_prototype_sim.launch" />
+
+        <node name="$(anon morse_launcher)" pkg="iarc7_simulator"
```
Comment (amiller27) [inline launch/morse.launch]: The two nodes here are exactly the same, can we put them outside these group statements?

File: sim/default.py (hunk 3 above)
Comment (satorinin) [reply sim/default.py]: fixed in 11056f1

**Comment by satorinin** [comment]
All current recommendations have been implemented.

**Comment by amiller27** [comment]
Is this tested after the changes?

**Comment by satorinin** [comment]
Testing now

**Comment by satorinin** [comment]
Takeoff_translate_land works with v1.1. v1.5 works, but it's supporting code in motion is buggy on my computer. Everything works when launching solely from morse.launch

**Comment by amiller27** [comment]
Has someone else tested v1.5?  Most of the new stuff is for v1.5, right?

**Comment by aftersomemath** [comment]
Yes, the tune in the sim is terrible but everything appears to work.

**Review by amiller27 — APPROVED** [review]

**Pull request merged**